thermalnetwork/heat_pump.py

```diff
@@ -1,29 +1,20 @@
 from thermalnetwork.base_component import BaseComponent
-from thermalnetwork.enums import ComponentType
+from thermalnetwork.enums import ComponentType, HPType
 
 
-class HeatPump(BaseComponent):
-    def __init__(self, data: dict) -> None:
-        super().__init__(data["name"], ComponentType.HEATPUMP)
-        props = data["properties"]
-        self.cop_c = props["cop_c"]
-        self.cop_h = props["cop_h"]
-
-    def calc_src_side_load(self, space_load: float) -> float:
-        """
-        Computes the loads on the source-side of the heat pump
-
-        :param space_load: space heating or cooling load. positive indicates heating, negative indicates cooling
-        :returns: heat extraction or heat rejection loads on source-side of heat pump
-        :rtype: float
-        """
-
-        if space_load >= 0.0:
-            # heating load
-            return space_load * (1 - 1 / self.cop_h)
-        else:
-            # cooling load
-            return space_load * (1 + 1 / self.cop_c)
+class HP(BaseComponent):
+    def __init__(self, hp_name: str, cop: float, hp_type: HPType):
+        super().__init__(hp_name, ComponentType.HEATPUMP)
+        self.hp_type = hp_type.strip().upper()
+        self.cop = cop
 
     def get_loads(self, loads: list[float]):
         return [self.calc_src_side_load(x) for x in loads]
+
+    def calc_src_side_load(self, load):
+        if self.hp_type == HPType.COOLING.name:
+            return load * (1 + 1 / self.cop)
+        elif self.hp_type in [HPType.HEATING.name, HPType.DHW.name]:
```

Comment:
I guess this works... `Type.TypeName.name` converts to string representation of the `TypeName`. I think you can just drop `.name` and compare types directly, which should be marginally more efficient since you're comparing `int`s, I think. If it works, then leave it and we can change it later.

Reply:
Yes, this current way works. If I remove the `.name` suffix then it raises the error a few lines below, which looks like this:
`ValueError: HEATING is not a valid heat pump type.`

I'll leave it as is for now. Happy to make it better whenever we can!